File: Baekjoon/누적합/11659.py
# https://www.acmicpc.net/problem/11659

import sys
input = sys.stdin.readline

n, m = map(int, input().split())
nums = list(map(int, input().split()))
idx_total = [0]
total = 0

for idx in range(n):
    total += nums[idx]
    idx_total.append(total)


for _ in range(m):
    i, j = map(int, input().split())
    print(idx_total[j] - idx_total[i - 1])

File: Baekjoon/누적합/11659.cpp
// https://www.acmicpc.net/problem/11659

#include <iostream>
#include <vector>
using namespace std;

int N, M;
int arr[100001];

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> N >> M;
    for (int i = 1; i <= N; i++) {
        cin >> arr[i];
        arr[i] += arr[i - 1];
    }

    while (M--) {
        int i, j;
        cin >> i >> j;
        cout << arr[j] - arr[i - 1] << '\n';
    }

    return 0;
}